### ISSUE TITLE
Assertion failed: nbytes == sizeof (dummy) 

### ISSUE THREAD

**Issue by rasviitanen** [zeromq/libzmq#4065]
*Please use this template for reporting suspected bugs or requests for help.*

# Issue description
Occationally `signaller.cpp` throws: Assertion failed: nbytes == sizeof (dummy)  on line 190
I am looking for list of possible reasons for why the assertion might fail.
I understand that there are previous issues for this, but they all seem to revolve around not using sockets on different threads. In my case, I don't.

# Environment
I use bindings from https://github.com/erickt/rust-zmq.

* libzmq version (commit hash if unreleased): 4.3.2
* OS: Windows 10

# Minimal test code / Steps to reproduce the issue
* I use a combination of PUSH/PULLL sockets (single sockets are thread-local and never sent/used on different threads but the one they were created on).

* My context is cloned between different threads and reused to create sockets.

* I don't formally call "disconnect".

* It occurs (but very seldom) when running our test-suite (tests runs in parallel)

Using this minimal example, I have been able to reproduce it (but only if i run other tests in the background (doing kinda the same thing)):

```rust
#[tokio::test]
async fn my_test() -> Result<()> {
        let ctx = zmq::Context::new();
        let addr = "tcp://127.0.0.1:48793".to_string();
        let n_sockets = 25usize;

        let ctx_clone = ctx.clone();
        let addr_clone = addr.clone();
        // Spawn a sink
        std::thread::spawn(move || {
            let pull_socket = ctx_clone.socket(zmq::PULL).unwrap();
            pull_socket.bind(&addr_clone).unwrap();

            for _ in 0..n_sockets {
                assert!(pull_socket.recv_msg(0).is_ok());
            }
        });

        let mut tasks = Vec::new();
        for _ in 0..n_sockets {
            let ctx = ctx.clone();
            let addr = addr.clone();
            let task = tokio::spawn(async move {
                let push_socket = ctx.socket(zmq::PUSH).unwrap();
                push_socket.connect(&addr).unwrap();
                push_socket.send(Vec::new(), 0).unwrap();
            });
            tasks.push(task);
        }

        for task in tasks {
            task.await?;
        }

        Ok(())
}
```

# What's the actual result? (include assertion message & call stack if applicable)
Assertion failed: nbytes == sizeof (dummy) (vendor\src\signaler.cpp:190)

# What's the expected result?
Assertion passed :)

**Comment by hoditohod** [comment]
Please have a look at this issue: https://github.com/zeromq/libzmq/issues/3756

This is a regression in 4.3.2, and as I understand it's fixed in 4.3.3.

**Comment by rasviitanen** [comment]
@hoditohod Cool, it seems to work now that we run 4.3.3, thanks!
